airflow/providers/amazon/aws/hooks/redshift_data.py

```diff
@@ -46,3 +48,73 @@ class RedshiftDataHook(AwsGenericHook["RedshiftDataAPIServiceClient"]):
     def __init__(self, *args, **kwargs) -> None:
         kwargs["client_type"] = "redshift-data"
         super().__init__(*args, **kwargs)
+
+    def execute_query(
+        self,
+        database: str,
+        sql: str | list[str],
+        cluster_identifier: str | None = None,
+        db_user: str | None = None,
+        parameters: list | None = None,
+        secret_arn: str | None = None,
+        statement_name: str | None = None,
+        with_event: bool = False,
+        await_result: bool = True,
```

Comment:
By convention, `wait_for_completion` is usually used as name for this kind of flag

Reply:
This was the flag name in the existing code, should we really change it?

Reply:
Oh yeah good point. We can either rename it but it has to go through deprecation pattern first (since it is breaking change) or leave it as is. I am fine keeping it as is

Reply:
Yes please lets rename. better to stay consistent